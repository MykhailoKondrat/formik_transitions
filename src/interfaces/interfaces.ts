import { FieldInputProps, FieldMetaProps } from "formik";

interface InputMeta {
  field?: FieldInputProps<string | undefined>;
  meta?: FieldMetaProps<string | undefined>;
}
export interface MyInput extends InputMeta {
  id?: string;
  placeholder: string;
  name: string;
  label: string;
}
export interface MySelect {
  id?: string;
  name: string;
  label: string;
  children?: React.ReactNode;
}

export interface InitialValues {
  firstName: string | undefined;
  mySelect: string | undefined;
}
