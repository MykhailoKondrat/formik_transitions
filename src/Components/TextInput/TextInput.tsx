import React from "react";
import TextField from "@material-ui/core/TextField";
import { MyInput } from "../../interfaces/interfaces";

const TextInput: React.FC<MyInput> = ({
  label,
  field,
  meta,
  ...props
}: MyInput) => {
  return (
    <>
      <TextField
        label={label}
        type="text"
        {...field}
        {...props}
        error={meta?.touched && meta.error ? true : undefined}
        helperText={meta?.error}
      />
    </>
  );
};

export default TextInput;
