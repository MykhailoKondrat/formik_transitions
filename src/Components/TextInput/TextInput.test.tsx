import TextInput from "./TextInput";
import { shallow } from "enzyme";
import { MyInput } from "../../interfaces/interfaces";
import React from "react";
import * as Adapter from "../../../setupTests";
const adapter = Adapter;

describe("<TextInput/>", () => {
  it("Match Snapshot", () => {
    const mockProps: MyInput = {
      label: "label",
      name: "name",
      placeholder: "placeholder",
      meta: {
        touched: false,
        error: "",
        initialError: "",
        initialTouched: false,
        initialValue: "",
        value: "",
      },
      field: {
        value: "",
        checked: false,
        onChange: jest.fn(),
        onBlur: jest.fn(),
        multiple: undefined,
        name: "firstName",
      },
    };

    expect(
      shallow(
        <TextInput
          label="label"
          name="name"
          placeholder="placeholder"
          meta={mockProps.meta}
        />
      )
    ).toMatchSnapshot();
    expect(true).toBe(true);
  });
});
