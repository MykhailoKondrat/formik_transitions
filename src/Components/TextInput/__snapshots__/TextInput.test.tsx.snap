// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput/> Match Snapshot 1`] = `
<Fragment>
  <WithStyles(ForwardRef(TextField))
    helperText=""
    label="label"
    name="name"
    placeholder="placeholder"
    type="text"
  />
</Fragment>
`;
